foreshadow/console.py

```diff
@@ -0,0 +1,244 @@
+            y_search = search_intents(y_train)
+            print("Searching over valid intent space for y data")
+
+        # If level 3 also do model parameter search with AutoEstimator
+        # Input time limit into Foreshadow to be passed into AutoEstimator
+
+        fs = Foreshadow(
+            X_preprocessor=X_search,
+            y_preprocessor=y_search,
+            estimator=get_method(cargs.method, X_train),
+            optimizer=GridSearchCV,
+        )
+
+    elif cargs.level == 3:
+        # Default intent and advanced model search using 3rd party AutoML
+
+        fs = Foreshadow(
+            estimator=AutoEstimator(
+                estimator_kwargs={"max_time_mins": cargs.time}
+            )
+        )
+
+    else:
+        raise ValueError("Invalid Level. Levels 1 - 3 supported.")
+
+    return fs, X_train, y_train, X_test, y_test
+
+
+def execute_model(fs, X_train, y_train, X_test, y_test):
+    """ Executes the model produced by generate_model()
+    and exports the data to json as well as returning the
+    exported json object containing the results and the serialized
+    Foreshadow object. Also prints simple model accuracy metrics.
+    """
+
+    print("Fitting final model...")
+    fs.fit(X_train, y_train)
+
+    print("Scoring final model...")
+    score = fs.score(X_test, y_test)
+
+    print("Final Results: ")
+    print(score)
+
+    # Store final results
+    all_results = {
+        "X_Model": fs.X_preprocessor.serialize(),
+        "X_Summary": fs.X_preprocessor.summarize(X_train),
+        "y_Model": fs.y_preprocessor.serialize(),
+        "y_summary": fs.y_preprocessor.summarize(y_train),
+    }
+
+    with open("model.json", "w") as outfile:
+        json.dump(all_results, outfile)
+
+    print(
+        "Results of model fiting have been saved to model.json."
+        "Refer to docs to read and process."
+    )
+
+    return all_results
+
+
+def cmd():  # pragma: no cover
+    """ Entry point to foreshadow via console command. Uncovered as
+    this function only serves to be executed manually.
+    """
+
+    model = generate_model(sys.argv[1:])
+    execute_model(*model)
+
+
+def get_method(arg, y_train):
+    """ Function to determine what estimator to use given a set of X data
+    and a passed argument referencing an sklearn Estimator class.
+    """
+
+    if arg is not None:
+        try:
+            mod = __import__(
+                "sklearn.linear_model", globals(), locals(), ["object"], 0
+            )
+            cls = getattr(mod, arg)
+            return cls()
+        except Exception:
+            raise ValueError(
+                "Invalid method. {} is not a valid "
+                "estimator from sklearn.linear_model".format(arg)
+            )
+
+    else:
+        return (
+            LinearRegression()
+            if determine_problem_type(y_train) == "regression"
+            else LogisticRegression()
+        )
+
+
+def search_intents(data):
+
+    proc = Preprocessor()
```

Comment:
Here we are treating X and y the same. Shall we set y_var properly when creating the Preprocessor?